lib/classes.rb

```diff
@@ -0,0 +1,130 @@
+class Player
+  attr_reader :name
+  attr_accessor :piece
+  attr_accessor :score
+  def initialize(name, piece = ' ', score = 0)
+    @name = name
+    @score = score
+    @piece = piece
+  end
+
+  def change_score
+    @score += 1
+  end
+end
+
+class Match
+  SEPARATOR = '----------'.freeze
+
+  def initialize(matches = 0)
+    @hash = { 1 => 1, 2 => 2, 3 => 3, 4 => 4, 5 => 5, 6 => 6, 7 => 7, 8 => 8, 9 => 9 }
+    @turn = 0
+    @matches = matches
+  end
+
+  def who_is_first?(player_one, player_two)
+    value = %w[X O]
+    select = Random.new.rand(2)
+    if select.zero?
+      player_one.piece = value[0]
+      player_two.piece = value[1]
+      player_one.name
+    else
+      player_one.piece = value[1]
+      player_two.piece = value[0]
+      player_two.name
+    end
+  end
+
+  def display_board(board_hash)
+    board_hash['row_one'].call(@hash)
+    board_hash['separator'].call(SEPARATOR)
+    board_hash['row_two'].call(@hash)
+    board_hash['separator'].call(SEPARATOR)
+    board_hash['row_three'].call(@hash)
+  end
+
+  def place_move(player_one, puts_hash, board_hash)
+    move = 0
+    until ((1..9).include? move) && (@hash.value? move)
+      puts_hash['another_num'].call
+      move = gets.chomp
+      exit if move == 'exit'
+      move = move.to_i
+    end
+    @hash[move] = player_one.piece
+    display_board(board_hash)
+  end
+
+  def next_turn(*args, puts_hash, board_hash)
+    @turn = 0
+    while @turn < 9
+      @turn += 1
+      if @turn.odd?
+        player_turn = args[0..-1].detect { |a| a.piece == 'X' }
+        player_next = args[0..-1].detect { |a| a.piece == 'O' }
+      else
+        player_turn = args[0..-1].detect { |a| a.piece == 'O' }
+        player_next = args[0..-1].detect { |a| a.piece == 'X' }
+      end
+      place_move(player_turn, puts_hash, board_hash)
+      check_winner(player_turn, player_next, puts_hash, board_hash) if @turn >= 5
+      puts_hash['your_turn'].call(player_next.name) if @turn < 9
+    end
+    puts_hash['draw'].call
+    keep_playing?(player_turn, player_next, puts_hash, board_hash)
+  end
```

Comment:
* [ ] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing).([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- This method has many responsibilities even though it should only change the player's turn it looks like it's handling the main gameplay placing moves, validating the winner, and also printing results on screen, the content of this whole method belongs in the loop that should exist in the main.rb file, so let's move it there.